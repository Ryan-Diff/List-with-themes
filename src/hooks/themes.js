import { useContext, useEffect, useState } from 'react';
import { ThemeContext } from '../context/ThemeContext';
import styles from '../components/theme-changer/Theme.css';

export const useThemePicker = () => {
    const { themeType, toggle } = useContext(ThemeContext);
    const className = styles[themeType];

    return { themeType, toggle, className };
};
