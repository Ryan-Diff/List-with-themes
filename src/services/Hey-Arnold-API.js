export const getCharacters = async(page =1) => {
    const res = await fetch(`https://hey-arnold-api.herokuapp.com/api/v1/characters?page=${page}`);
    const json = await res.json();

    if(!res.ok) throw 'Sorry about it';

    return json.map(character => ({
        id: character._id,
        name: character.name,
        image: character.image
    }));
};
