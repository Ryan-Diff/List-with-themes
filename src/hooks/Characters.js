import { useEffect, useState } from 'react';
import { getCharacters } from "../services/Hey-Arnold-API";

export const useCharacters = () => {
    const [characters, setCharacters] = useState([]);

    useEffect(() => {
        getCharacters()
        .then(fetchedCharacters => setCharacters(fetchedCharacters));
    }, []);

    return characters;
};