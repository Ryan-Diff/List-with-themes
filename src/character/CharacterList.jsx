import React from 'react';
import PropTypes from 'prop-types';
import CharacterItem from './CharacterItem';
import { useCharacters } from '../hooks/Characters';

const CharacterList = () => {
    const characters = useCharacters();
    
    const characterElements = characters.map(character => (
        <li key={character.id}>
            <CharacterItem {...character} />
        </li>
    ));

    return (
        <ul data-testid='characters'>
            {characterElements}
        </ul>
    );
};

export default CharacterList;
    
