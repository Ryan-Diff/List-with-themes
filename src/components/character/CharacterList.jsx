import React from 'react';
import CharacterItem from './CharacterItem';
import { useCharacters } from '../../hooks/Characters';
import { useThemePicker } from '../../hooks/themes';
import styles from '../theme-changer/Theme.css';

const CharacterList = () => {
    const characters = useCharacters();
    const { className } = useThemePicker();
    
    const characterElements = characters.map(character => (
        <li key={character.id}>
            <CharacterItem {...character} />
        </li>
    ));

    return (
        <ul data-testid='characters' className={`${styles.CharacterList} ${className}`}>
            {characterElements}
        </ul>
    );
};

export default CharacterList;
    
