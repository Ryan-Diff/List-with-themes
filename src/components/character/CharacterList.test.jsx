import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { getCharacters } from '../../services/Hey-Arnold-API';
import { useThemePicker } from '../../hooks/themes';

jest.mock('../../services/Hey-Arnold-API');
jest.mock('../../hooks/themes');

describe('CharacterList component', () => {
  afterEach(() => cleanup());

  it('Renders a list of Characters', () => {
    useThemePicker.mockResolvedValue('dark');
    getCharacters.mockResolvedValue([
      {
        'name': 'Ryan',
        'image': 'Ryan.png'
      }
    ])
    const { asFragment } = render(<CharacterList />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 