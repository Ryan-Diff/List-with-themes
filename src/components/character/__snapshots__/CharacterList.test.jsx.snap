// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component Renders a list of Characters 1`] = `
<DocumentFragment>
  <ul
    class="CharacterList undefined"
    data-testid="characters"
  />
</DocumentFragment>
`;
